skpro/regression/gradient_boosting/_ngboost.py

```diff
@@ -0,0 +1,167 @@
+                             set aside as validation data for early stopping.
+        early_stopping_rounds: The number of consecutive
+            boosting iterations during which the
+            loss has to increase before the algorithm stops early.
+            Set to None to disable early stopping and validation
+            None enables running over the full data set.
+
+
+    Returns
+    -------
+        An NGBRegressor object that can be fit.
+    """
+
+    def __init__(
+        self,
+        Dist=Normal,
+        Score=LogScore,
+        Base=default_tree_learner,
+        natural_gradient=True,
+        n_estimators=500,
+        learning_rate=0.01,
+        minibatch_frac=1.0,
+        col_sample=1.0,
+        verbose=True,
+        verbose_eval=100,
+        tol=1e-4,
+        random_state=None,
+        validation_fraction=0.1,
+        early_stopping_rounds=None,
+    ):
+        self.Dist = Dist
+        self.Score = Score
+        self.Base = Base
+        self.Manifold = manifold(Score, Dist)
+        self.natural_gradient = natural_gradient
+        self.n_estimators = n_estimators
+        self.learning_rate = learning_rate
+        self.minibatch_frac = minibatch_frac
+        self.col_sample = col_sample
+        self.verbose = verbose
+        self.verbose_eval = verbose_eval
+        self.init_params = None
+        self.n_features = None
+        self.base_models = []
+        self.scalings = []
+        self.col_idxs = []
+        self.tol = tol
+        self.random_state = check_random_state(random_state)
+        self.best_val_loss_itr = None
+        self.validation_fraction = validation_fraction
+        self.early_stopping_rounds = early_stopping_rounds
+
+        super().__init__()
+
+    def _fit(self, X, y):
+        """Fit regressor to training data.
+
+        Writes to self:
+            Sets fitted model attributes ending in "_".
+
+        Parameters
+        ----------
+        X : pandas DataFrame
+            feature instances to fit regressor to
+        y : pandas DataFrame, must be same length as X
+            labels to fit regressor to
+
+        Returns
+        -------
+        self : reference to self
+        """
+        self.ngb = NGBRegressor(
+            Dist=self.Dist,
+            Score=self.Score,
+            Base=default_tree_learner,
+            natural_gradient=True,
+            n_estimators=self.n_estimators,
+            learning_rate=self.learning_rate,
+            minibatch_frac=self.minibatch_frac,
+            col_sample=self.col_sample,
+            verbose=self.verbose,
+            verbose_eval=self.verbose_eval,
+            tol=self.tol,
+            random_state=self.random_state,
+            validation_fraction=self.validation_fraction,
+            early_stopping_rounds=self.early_stopping_rounds,
+        )
+        self.ngb.fit(X, y)
+        return self
+
+    def _predict(self, X):
+        return self.ngb.predict(X)
+
+    def _pred_dist(self, X):
```

Comment:
good start! This is, of course, where the crux of the matter will sit.

In `_predict_proba`, you'll need to convert an `ngboost` distribution int equivalent `skpro` distributions. I recommend using adaptor/converter patterns.

Reply:
Sorry was a bit busy last couple of days. By converting do you mean like using something like `SklearnProbaReg` adapter from `_sklearn_proba` but for `ngboost` I'd have to make one. Do I understand correctly?

Reply:
> Do I understand correctly?

Yes - I would do it entirely on the level of distributions first, i.e., transforming the `ngboost` return type into an `skpro` `BaseDistribution`.

My guess would be, you'll not need an extra adapter on the level of regressor, if you call the conversion from `_predict_proba`.